tests/backup/backup_helper.go

```diff
@@ -1337,3 +1393,87 @@ func CreateCustomRestoreWithPVCs(restoreName string, backupName string, namespac
 	}
 	return deployment.Name, nil
 }
+
+// GetOrdinalScheduleBackupName returns the name of the schedule backup at the specified ordinal position for the given schedule
+func GetOrdinalScheduleBackupName(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+	}
+	if ordinal < 1 {
+		return "", fmt.Errorf("the ordinal value [%d] for schedule backups with schedule name [%s] is invalid. valid values range from 1 to [%d]", ordinal, scheduleName, len(allScheduleBackupNames))
+	}
+	if ordinal > len(allScheduleBackupNames) {
+		return "", fmt.Errorf("schedule backups with schedule name [%s] have not been created up to the provided ordinal value [%d]", scheduleName, ordinal)
+	}
+	return allScheduleBackupNames[ordinal-1], nil
+}
+
+// GetFirstScheduleBackupName returns the name of the first schedule backup for the given schedule
+func GetFirstScheduleBackupName(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups found for schedule %s", scheduleName)
+	}
+	return allScheduleBackupNames[0], nil
+}
+
+// GetLatestScheduleBackupName returns the name of the latest schedule backup for the given schedule
+func GetLatestScheduleBackupName(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups found for schedule %s", scheduleName)
+	}
+	return allScheduleBackupNames[len(allScheduleBackupNames)-1], nil
+}
+
+// GetOrdinalScheduleBackupUID returns the uid of the schedule backup at the specified ordinal position for the given schedule
+func GetOrdinalScheduleBackupUID(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	allScheduleBackupUids, err := Inst().Backup.GetAllScheduleBackupUIDs(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupUids) == 0 {
+		return "", fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+	}
+	if ordinal < 1 {
```

Comment:
same

Reply:
If the ordinal value is less than 1, I am using the length of allScheduleBackupNames to indicate the valid range of values in the log.

Reply:
Made sure to check ordinal value before enumerating backups.